addons/html_editor/static/src/core/delete_plugin.js

```diff
@@ -134,20 +137,103 @@ export class DeletePlugin extends Plugin {
     // --------------------------------------------------------------------------
 
     /**
-     * @param {EditorSelection} [selection]
+     * Deletes the contents in the current selection, if any.
+     * No history step is added.
+     * Use it alongside other command to compose a step.
      */
-    deleteSelection(selection = this.shared.getEditableSelection()) {
+    deleteSelection() {
         // @todo @phoenix: handle non-collapsed selection around a ZWS
         // see collapseIfZWS
+        const range = this.getNormalizedRange();
+        if (
+            range &&
+            !range.collapsed &&
+            closestElement(range.commonAncestorContainer).isContentEditable
+        ) {
+            this.deleteNonCollapsedRange(range);
+        }
+    }
 
-        // Normalize selection
-        selection = this.shared.setSelection(selection);
-
-        if (selection.isCollapsed) {
+    /**
+     * If the selection in non-collapsed, deletes its content.
+     * Otherwise, deletes a character/word/line in the given direction.
+     * Adds a history step. Use it as a standalone command.
+     *
+     * @param {"backward"|"forward"} direction
+     * @param {"character"|"word"|"line"} granularity
+     */
+    delete(direction, granularity) {
+        const range = this.getNormalizedRange();
+        if (!range || !closestElement(range.commonAncestorContainer).isContentEditable) {
             return;
         }
 
-        let range = this.adjustRange(selection, [
+        if (!range.collapsed) {
+            this.deleteNonCollapsedRange(range);
+        } else if (direction === "backward") {
+            this.deleteBackward(range, granularity);
+        } else if (direction === "forward") {
+            this.deleteForward(range, granularity);
+        } else {
+            throw new Error("Invalid direction");
+        }
+
+        this.dispatch("ADD_STEP");
+    }
+
+    // --------------------------------------------------------------------------
+    // Selection normalization
+    // --------------------------------------------------------------------------
+
+    /**
+     * Gets the **document** selection, normalizes it and returns a Range.
+     * It does not use getEditableSelection from the SelectionPlugin because:
+     * - it is normalized around contenteditable=false, making it impossible to
+     *   know if the original selection was inside an uneditable zone,
+     * - it is not normalized deep.
```

Comment:
Then what kind of normalization does it do? I'm asking because, as a new reader of the doc, I have no way to know just by reading the doc.

Reply:
Indeed. I have updated the doc!